integration-tests/actions/vrf/vrfv2plus/setup_steps.go

```diff
@@ -0,0 +1,478 @@
+	env, err := test_env.NewCLTestEnvBuilder().
+		WithTestInstance(t).
+		WithTestConfig(&testConfig).
+		WithCustomCleanup(cleanupFn).
+		Build()
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("%s, err: %w", "error creating test env", err)
+	}
+	coordinator, err := env.ContractLoader.LoadVRFCoordinatorV2_5(*commonExistingEnvConfig.CoordinatorAddress)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("%s, err: %w", "error loading VRFCoordinator2_5", err)
+	}
+	linkToken, err := env.ContractLoader.LoadLINKToken(*commonExistingEnvConfig.LinkAddress)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("%s, err: %w", "error loading LinkToken", err)
+	}
+	err = vrfcommon.FundNodesIfNeeded(ctx, commonExistingEnvConfig, env.EVMClient, l)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("err: %w", err)
+	}
+	vrfContracts := &vrfcommon.VRFContracts{
+		CoordinatorV2Plus: coordinator,
+		VRFV2PlusConsumer: nil,
+		LinkToken:         linkToken,
+		BHS:               nil,
+	}
+
+	vrfKey := &vrfcommon.VRFKeyData{
+		VRFKey:            nil,
+		EncodedProvingKey: [2]*big.Int{},
+		KeyHash:           common.HexToHash(*commonExistingEnvConfig.KeyHash),
+	}
+	return vrfContracts, vrfKey, env, nil
+}
+
+func SetupSubsAndConsumersForExistingEnv(
+	env *test_env.CLClusterTestEnv,
+	coordinator contracts.VRFCoordinatorV2_5,
+	linkToken contracts.LinkToken,
+	numberOfConsumerContractsToDeployAndAddToSub int,
+	numberOfSubToCreate int,
+	testConfig tc.TestConfig,
+	l zerolog.Logger,
+) ([]*big.Int, []contracts.VRFv2PlusLoadTestConsumer, error) {
+	var (
+		subIDs    []*big.Int
+		consumers []contracts.VRFv2PlusLoadTestConsumer
+		err       error
+	)
+	if *testConfig.VRFv2Plus.General.UseExistingEnv {
+		commonExistingEnvConfig := testConfig.VRFv2Plus.ExistingEnvConfig.ExistingEnvConfig
+		if *commonExistingEnvConfig.CreateFundSubsAndAddConsumers {
+			consumers, subIDs, err = SetupNewConsumersAndSubs(
+				env,
+				coordinator,
+				testConfig,
+				linkToken,
+				numberOfConsumerContractsToDeployAndAddToSub,
+				numberOfSubToCreate,
+				l,
+			)
+			if err != nil {
+				return nil, nil, fmt.Errorf("err: %w", err)
+			}
+		} else {
+			consumer, err := env.ContractLoader.LoadVRFv2PlusLoadTestConsumer(*commonExistingEnvConfig.ConsumerAddress)
+			if err != nil {
+				return nil, nil, fmt.Errorf("err: %w", err)
+			}
+			consumers = append(consumers, consumer)
+			var ok bool
+			subID, ok := new(big.Int).SetString(*testConfig.VRFv2Plus.ExistingEnvConfig.SubID, 10)
+			if !ok {
+				return nil, nil, fmt.Errorf("unable to parse subID: %s %w", *testConfig.VRFv2Plus.ExistingEnvConfig.SubID, err)
+			}
+			subIDs = append(subIDs, subID)
+		}
+	} else {
+		consumers, subIDs, err = SetupNewConsumersAndSubs(
+			env,
+			coordinator,
+			testConfig,
+			linkToken,
+			numberOfConsumerContractsToDeployAndAddToSub,
+			numberOfSubToCreate,
+			l,
+		)
+		if err != nil {
+			return nil, nil, fmt.Errorf("err: %w", err)
+		}
```

Comment:
Remember the comment where I suggested to reorganize the code by flipping the logic? This can be applied here as well.